### DISCUSSION TITLE
How to integrate Admin UI?

### DISCUSSION THREAD

**Discussion by nursimaaigoritma** [miguelgrinberg/Flask-SocketIO#2021] [Q&A] [answered]
Hello, I want to integrate Socketio Admin UI to our flask socketio app. Since it is not mentioned in the flask socketio docs, I look at the python-socketio and found [this](https://python-socketio.readthedocs.io/en/stable/server.html#monitoring-and-administration),

```
import os
import socketio

sio = socketio.Server(cors_allowed_origins=[
    'http://localhost:5000',
    'https://admin.socket.io',
])
sio.instrument(auth={
    'username': 'admin',
    'password': os.environ['ADMIN_PASSWORD'],
})
```

I try to access the server in our app
```
SOCKET_CONFIG = {
    'async_mode': 'threading',
    'cors_allowed_origins': "*",
    'manage_session': False,
    'logger': True,
    'ping_timeout':60,
    'ping_interval':120
}
from flask_socketio import SocketIO
socketio = SocketIO(**SOCKET_CONFIG)
app = Flask(__name__)
socketio.init_app(app)
socketio.server.instrument(auth=False)
```
But it gives the error `AttributeError: 'Server' object has no attribute 'instrument'`. Why couldn't it find the instrument method? I'd appreciate any directions on that. Thank you!

**Comment by miguelgrinberg** [comment, marked as answer]
Note that I haven't done the integration work on Flask-SocketIO yet, so at this point I really cannot offer support. That said, I'm not really sure why the `instrument()` method is not available. Are you sure you have the latest python-socketio package?

**Reply by nursimaaigoritma** [reply]
Thank you for your quick response. You're right, I upgraded to 5.10.0. And used the InstrumentedServer like you suggested [here](https://github.com/miguelgrinberg/python-socketio/pull/1164#issuecomment-1579246854). I can get the stats.
